lib/SPIRV/SPIRVToOCL20.cpp

```diff
@@ -135,11 +135,23 @@ void SPIRVToOCL20::visitCallSPIRVControlBarrier(CallInst *CI) {
           return cast<ConstantInt>(Args[I])->getZExtValue();
         };
         auto ExecScope = static_cast<Scope>(GetArg(0));
-        auto MScope = static_cast<Scope>(GetArg(1));
-        auto Sema = mapSPIRVMemSemanticToOCL(GetArg(2));
+        auto MemScope = static_cast<Scope>(GetArg(1));
+
         Args.resize(2);
-        Args[0] = getInt32(M, Sema.first);
-        Args[1] = getInt32(M, rmap<OCLScopeKind>(MScope));
+        if (auto Arg = dyn_cast<ConstantInt>(Args[2])) {
```

Comment:
This is accessing `Args[2]`, but there's `Args.resize(2)` a few lines above.

Reply:
Good catch, thanks! Should be fixed now